src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1019 @@
+        }
+      }
+      feature.setStyle(highlightStyle);
+    });
+  }
+
+  /**
+   * Selects the selected feature in the map and in the grid.
+   *
+   * @param {ol.MapBrowserEvent} olEvt The ol event.
+   */
+  onMapSingleClick = olEvt => {
+    const {
+      map,
+      selectStyle
+    } = this.props;
+
+    const selectedRowKeys = this.getSelectedRowKeys();
+
+    const selectedFeatures = map.getFeaturesAtPixel(olEvt.pixel, {
+      layerFilter: layerCand => layerCand === this._layer
+    }) || [];
+
+    selectedFeatures.forEach(selectedFeature => {
+      const key = this.props.keyFunction(selectedFeature);
+      if (selectedRowKeys.includes(key)) {
+        selectedFeature.setStyle(null);
+
+        const node = this.getRowFromFeatureId(selectedFeature.ol_uid);
+        if (node) {
+          node.setSelected(false);
+        }
+      } else {
+        selectedFeature.setStyle(selectStyle);
+
+        const node = this.getRowFromFeatureId(selectedFeature.ol_uid);
+        if (node) {
+          node.setSelected(true);
+        }
+      }
+    });
+  }
+
+  /**
+   * Removes the vector layer from the given map (if any).
+   */
+  deinitVectorLayer = () => {
+    const {
+      map
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    map.removeLayer(this._layer);
+  }
+
+  /**
+   * Unbinds the pointermove and click event handlers from the map (if given).
+   */
+  deinitMapEventHandlers = () => {
+    const {
+      map,
+      selectable
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    map.un('pointermove', this.onMapPointerMove);
+
+    if (selectable) {
+      map.un('singleclick', this.onMapSingleClick);
+    }
+  }
+
+  /**
+   * Returns the column definitions out of the attributes of the first
+   * given feature.
+   *
+   * @return {Array} The column definitions.
+   */
+  getColumnDefs = () => {
+    const {
+      attributeBlacklist,
+      features,
+      columnDefs,
+      selectable
+    } = this.props;
+
+    let columns = [];
+    const feature = features[0];
+
+    if (!(feature instanceof OlFeature)) {
+      return columns;
+    }
+
+    const props = feature.getProperties();
+
+    if (selectable) {
+      columns.push({
+        headerName: '',
+        checkboxSelection: true,
+        headerCheckboxSelection: true,
+        width: 28,
+        pinned: 'left',
+        lockPosition: true,
+        suppressMenu: true,
+        suppressSorting: true,
+        suppressFilter: true,
+        suppressResize: true,
+        suppressMovable: true
+      });
+    }
+
+    let index = -1;
+
+    Object.keys(props).forEach(key => {
+      if (attributeBlacklist.includes(key)) {
+        return;
+      }
+
+      if (props[key] instanceof OlGeomGeometry) {
+        return;
+      }
+
+      if (columnDefs[key].index !== undefined) {
+        index = columnDefs[key].index;
+      } else {
+        ++index;
+      }
```

Comment:
This means that if one column has an index, all of them should have an index, right? We should document that very cleanly

Reply:
I've added an appropriate comment on the `columnDefs` prop.